internal/controller/worker_controller_test.go

```diff
@@ -20,299 +22,252 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	temporaliov1alpha1 "github.com/DataDog/temporal-worker-controller/api/v1alpha1"
-	"github.com/DataDog/temporal-worker-controller/internal/controller/k8s.io/utils"
 )
 
 var (
-	testPodTemplate = v1.PodTemplateSpec{
-		ObjectMeta: metav1.ObjectMeta{
-			Labels: map[string]string{
-				"temporal.io/build-id": "123456789",
-			},
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name:  "main",
-					Image: "foo/bar@sha256:deadbeef",
-				},
-			},
-		},
-	}
-
 	testTemporalNamespace = "ns"
 )
 
-// Hash is always taken of Spec.Template for generating the buildID.
-func newTestWorkerSpec(replicas int32) *temporaliov1alpha1.TemporalWorkerDeploymentSpec {
-	return &temporaliov1alpha1.TemporalWorkerDeploymentSpec{
-		Replicas: &replicas,
-		Template: testPodTemplate,
-		WorkerOptions: temporaliov1alpha1.WorkerOptions{
-			TemporalNamespace: testTemporalNamespace,
-		},
-	}
-}
-
-// newTestDeployment creates a new Deployment object for testing purposes.
-// If buildID is empty, it will be computed from the desiredState.Template.
-func newTestDeployment(podSpec v1.PodTemplateSpec, desiredState *temporaliov1alpha1.TemporalWorkerDeploymentSpec, deploymentName string, buildID string) *appsv1.Deployment {
-	// If buildID is not provided, compute it from the template
-	if buildID == "" {
-		buildID = utils.ComputeHash(&desiredState.Template, nil)
+func TestGeneratePlan(t *testing.T) {
```

Comment:
The easiest way to review this change might be to just open the new file and read it as a whole first to understand the format, and then look at the diff.

The TLDR of the benefits here is that in this new structure:
- more of the real genstatus / genplan / utils code is reused, making this more representative
- fewer things need to be hardcoded
- there are ways to add different rollout and sunset policies for different test cases
- all of it is intended to be more readable and reusable than before!